csrc/executor.cpp

```diff
@@ -1891,4 +1893,267 @@ float FusionExecutor::runRtc(
+  //   output_aliases : [int];
+  //   input_aliases : [int];
+  //   outputs : [GlobalBufferInfo];
+  //   intermediates : [GlobalBufferInfo];
+  //   rand_offset : ulong;
+  // }
+  std::vector<int> output_aliases_fb;
+  std::vector<int> input_aliases_fb;
+  output_aliases_fb.reserve(data.output_to_input_aliases.size());
+  input_aliases_fb.reserve(data.output_to_input_aliases.size());
+  for (auto [out, in] : data.output_to_input_aliases) {
+    output_aliases_fb.push_back(out);
+    input_aliases_fb.push_back(in);
+  }
+
+  using fb_global_buffer_info =
+      flatbuffers::Offset<nvfuser::serde::GlobalBufferInfo>;
+  std::vector<fb_global_buffer_info> outputs_fb;
+  outputs_fb.reserve(data.outputs.size());
+  for (const auto& buffer : data.outputs) {
+    auto tv_iter = std::find(
+        kernel()->outputs().cbegin(), kernel()->outputs().cend(), buffer.tv);
+    auto tv_position = (tv_iter == kernel()->outputs().cend())
+        ? -1
+        : std::distance(kernel()->outputs().cbegin(), tv_iter);
+    outputs_fb.push_back(
+        serialize(builder, buffer, tv_position, true /* is_fusion_output */));
+  }
+
+  std::vector<fb_global_buffer_info> intermediates_fb;
+  intermediates_fb.reserve(data.intermediates.size());
+  for (const auto& buffer : data.intermediates) {
+    auto match_tv_predicate = [buffer_tv = buffer.tv](const kir::Allocate* a) {
+      return a->buffer() == buffer_tv;
+    };
+    auto tv_iter = std::find_if(
+        kernel()->summary().global_allocations.cbegin(),
+        kernel()->summary().global_allocations.cend(),
+        match_tv_predicate);
+    auto tv_position =
+        (tv_iter == kernel()->summary().global_allocations.cend())
+        ? -1
+        : std::distance(
+              kernel()->summary().global_allocations.cbegin(), tv_iter);
+    intermediates_fb.push_back(
+        serialize(builder, buffer, tv_position, false /* is_fusion_output */));
+  }
+
+  return CreateExecutorEntryDirect(
+      builder,
+      data.init,
+      data.launch_params.serialize(builder),
+      &output_aliases_fb,
+      &input_aliases_fb,
+      &outputs_fb,
+      &intermediates_fb,
+      data.rand_offset);
+}
+
+flatbuffers::Offset<serde::GlobalBufferInfo> FusionExecutor::serialize(
+    flatbuffers::FlatBufferBuilder& builder,
+    const GlobalBufferInfo& data,
+    int64_t tv_position,
+    bool is_fusion_output) const {
+  // table GlobalBufferInfo {
+  //  tv : long;
+  //  sizes : [long];
+  //  strides : [long];
+  //  type : DataType;
+  //  zero_init : bool;
+  //  is_profile_buffer : bool;
+  //  is_fusion_output : bool;
+  // }
+  return serde::CreateGlobalBufferInfoDirect(
+      builder,
+      tv_position,
+      &data.sizes,
+      &data.strides,
+      serde::mapToSerdeDtype(data.type),
+      data.zero_init,
+      data.is_profile_buffer,
+      is_fusion_output);
+}
+
+void FusionExecutor::deserialize(
+    const serde::FusionExecutor* buffer,
+    Fusion* fusion) {
+  // table FusionExecutor {
```

Comment:
why does this one disagrees from the definition below in csrc/serde/fusion_cache.fbs? Specifically we added:
```
  //  configured_device_smem : ulong;
  //  maybe_available_dynamic_smem : ulong;
```

Reply:
I must have forgot to update this comment. `fusion-cache.fbs` is the correct reference.